### PR TITLE
feat: upgrade to eslint 6

### PR DESCRIPTION
**Pull request by sodatea**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Underlying tools
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**Other information:**

### CODE REVIEW

File: packages/@vue/cli-plugin-eslint/generator/index.js [hunk 1]
```diff
@@ -2,49 +2,75 @@ const fs = require('fs')
...
+  if (api.hasPlugin('typescript')) {
+    Object.assign(pkg.devDependencies, {
+      '@vue/eslint-config-typescript': '^5.0.1',
+      '@typescript-eslint/eslint-plugin': '^2.10.0',
+      '@typescript-eslint/parser': '^2.10.0'
+    })
+    if (config !== 'prettier') {
+      // for any config other than `prettier`,
+      // typescript ruleset should be appended to the end of the `extends` array
+      eslintConfig.extends.push('@vue/typescript/recommended')
```
Comment (sodatea) [inline packages/@vue/cli-plugin-eslint/generator/index.js]: Not sure whether we should use `@vue/typescript/recommended` or `@vue/typescript` here.

File: packages/@vue/cli-plugin-eslint/generator/index.js [hunk 2]
```diff
@@ -130,7 +149,8 @@ module.exports.hooks = (api) => {
   })
 }
 
-const applyTS = module.exports.applyTS = api => {
+// exposed for the typescript plugin
+module.exports.applyTS = api => {
```
Comment (pksunkara) [inline packages/@vue/cli-plugin-eslint/generator/index.js]: Where is this being used?

**Comment by pksunkara** [comment]
I gave it a basic look. Looks good.

File: packages/@vue/cli-plugin-eslint/generator/index.js (hunk 2 above)
Comment (sodatea) [reply packages/@vue/cli-plugin-eslint/generator/index.js]: Used here: https://github.com/vuejs/vue-cli/blob/1c269d22f5c15e313966db8ce7db23d9b0758d8f/packages/%40vue/cli-plugin-typescript/generator/index.js#L63-L79

**Comment by sodatea** [comment]
😅Finally, all the tests are passed!
I think this PR is good to go now… I'd like to write migration instructions in a separate PR, as this one is already somehow messy…

**Pull request merged**